components/playlist/browser/resources/components/playlistFolder.tsx

```diff
@@ -145,34 +169,148 @@ export default function PlaylistFolder ({
     [playlist, selectedSet, editMode]
   )
 
+  const [draggedOrder, setDraggedOrder] = React.useState<
+    PlaylistItemMojo[] | null
+  >(null)
+
+  const sensors = useSensors(
+    useSensor(MouseSensor, {
+      activationConstraint: { distance: 40 }
+    }),
+    useSensor(PointerSensor, {
+      activationConstraint: { distance: 40 }
+    }),
+    useSensor(TouchSensor, {
+      activationConstraint: {
+        delay: 250,
+        tolerance: 40
+      }
+    })
+  )
+
   if (!playlist) {
     // After deleting a playlist from header, this could happen. In this case,
     // redirect to the index page.
     return <Redirect to='/' />
   }
 
-  const itemsToRender = lastPlayerState?.shuffleEnabled
-    ? lastPlayerState.currentList?.items ?? playlist?.items
-    : playlist?.items
+  const itemsToRender =
+    draggedOrder ??
+    (lastPlayerState?.shuffleEnabled
+      ? lastPlayerState.currentList?.items ?? playlist?.items
+      : playlist?.items)
 
-  return (
-    <>
-      {editMode === PlaylistEditMode.BULK_EDIT && (
-        <EditActionsContainer
-          playlistId={playlist.id!}
-          selectedIds={selectedSet}
-        />
-      )}
-      {itemsToRender.map(item => (
+  const getPlaylistItem = (
+    item: PlaylistItemMojo | undefined,
+    isDraggable: boolean,
+    shouldBeHidden: boolean
+  ) => {
+    if (!item) return null
+
+    if (!isDraggable) {
```

Comment:
Optional:

```tsx
const Item = isDraggable ? PlaylistItem : SortablePlaylistItem

return <Item ... />
```

Reply:
Oh, didn't realize that we can hold jsx elemet in a variable 👍